salt/loader.py

```diff
@@ -2260,3 +2312,20 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     return inner_decorator
+
+
+@contextmanager
+def catch_entry_points_exception(entry_point):
+    context = types.SimpleNamespace(exception_caught=False)
+    try:
+        yield context
+    except Exception as exc:  # pylint: disable=broad-except
+        context.exception_caught = True
+        entry_point_details = entrypoints.name_and_version_from_entry_point(entry_point)
+        log.error(
+            "Error processing Salt Extension %s(version: %s): %s",
+            entry_point_details.name,
+            entry_point_details.version,
+            exc,
+            exc_info_on_loglevel=logging.DEBUG,
+        )
```

Comment:
**question**: Just double checking - this *does* end out dropping the reference to the `exc` stack frames, right? i.e. this isn't introducing any memory leaks.

I'm pretty sure that this is good and we are **not** but I want to make sure I'm not overlooking anything :upside_down_face:

Reply:
once the log record get's formatted, these references are gone, if they are not, it's the stdlib to blame :grin: